backend/server/src/http/assets.rs

```diff
@@ -0,0 +1,162 @@
+    "fonts/latin-ext-i400.woff2": { hash },
+    "fonts/latin-ext-i700.woff2": { hash },
+    "fonts/latin-i400.woff2": { hash },
+    "fonts/latin-i700.woff2": { hash },
+    "fonts/vietnamese-400.woff2": { hash },
+    "fonts/vietnamese-700.woff2": { hash },
+    "fonts/vietnamese-i400.woff2": { hash },
+    "fonts/vietnamese-i700.woff2": { hash },
+};
+
+const INDEX_FILE: &str = "index.html";
+
+pub(crate) struct Assets {
+    assets: reinda::Assets,
+
+    // TODO: this is temporary until we update hyper to tokio 1.0. We need to
+    // explicitly store the runtime here to use the tokio 1.0 runtime inside of
+    // the handler functions as those are called within the hyper handler, which
+    // is a tokio 0.2 runtime.
+    runtime: tokio::runtime::Handle,
+}
+
+impl Assets {
+    pub(crate) async fn init(config: &config::Assets) -> Result<Self> {
+        // TODO: temporary
+        let runtime = tokio::runtime::Handle::current();
+
+        let mut path_overrides = HashMap::new();
+        path_overrides.insert("logo-large.svg".into(), config.logo.large.clone());
+        path_overrides.insert("logo-small.svg".into(), config.logo.small.clone());
+
+        let config = Config {
+            base_path: Some(config.internal.clone()),
+            path_overrides,
+            .. Config::default()
+        };
+
+        let assets = reinda::Assets::new(ASSETS, config).await
+            .context("failed to prepare asset files")?;
+        info!("Prepared {} assets", assets.asset_ids().count());
+
+        Ok(Self { assets, runtime } )
+    }
+
+    /// Responds with the asset identified by the given path. If there exists no
+    /// asset with `path` or `path` is `INDEX_FILE`, `None` is returned.
+    pub(crate) async fn serve(&self, path: &str) -> Option<Response> {
+        // The `index.html` here is not intended to be served directly. It is
+        // modified and sent on many other routes.
+        if path == INDEX_FILE {
+            return None;
+        }
+
+        // TODO: temporary
+        let _guard = self.runtime.enter();
+        let data = self.assets.get(path).await.unwrap_or_else(|e| {
+            panic!("failed to read asset '{}': {}", path, e);
+        })?;
+
+        // Prepare HTTP headers
+        let mut builder = Response::builder();
+        builder = builder.header("content-length", data.len());
+
+        // Mime type
+        let mime_guess = mime_guess::from_path(path).first();
+        if let Some(mime) = mime_guess {
+            builder = builder.header("content-type", mime.to_string())
+        }
+
+        // Caching header if the filename contains a content hash. We can unwrap
+        // the `lookup` call as we know from above that the path is valid.
+        if self.assets.asset_info(self.assets.lookup(path).unwrap()).is_filename_hashed() {
+            // This is one year in seconds.
+            builder = builder.header("cache-control", "public, max-age=31536000");
```

Comment:
Should we add `immutable` here? See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#caching_static_assets

Reply:
Seems reasonable